numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -329,6 +336,107 @@ array_implement_array_function(
 }
 
 
+/*
+ * Implements the __array_function__ protocol for a Python function, as described in
+ * in NEP-18. See numpy.core.overrides for a full docstring.
+ */
+NPY_NO_EXPORT PyObject *
+array_implement_array_function(
+    PyObject *NPY_UNUSED(dummy), PyObject *positional_args)
+{
+    PyObject *implementation, *public_api, *relevant_args, *args, *kwargs;
+
+    if (!intern_strings()) {
+        if (PyErr_Occurred()) {
```

Comment:
No need for this inner if, if `intern_strings` failed an exception is already set with a more explanatory message.

Reply:
Done.